### PR TITLE
fix: Relabel configs that drop if scheme is HTTP/HTTPS

### PR DESCRIPTION
**Pull request by skhalash**
## Description

Changes proposed in this pull request:

- Fix a bug resulting in the target-dropping mechanism not working properly if `prometheus.io/scheme` annotation is not set. This fix ensures that the target dropping functionality works as intended, regardless of the annotation's configuration

## Related Issues and Documents

Related issues:
https://github.com/kyma-project/telemetry-manager/issues/349

## Traceability

- [x] The PR is linked to a GitHub Issue.
- [x] New features have a milestone label set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub Issue has a respective `area` label.
- [x] The follow-up issues (if any) are linked in the `Related Issues` section.

## Testability

The feature is unit-tested:

- [x] Yes.
- [ ] No, because unit tests are not needed.
- [ ] No, because of ...

The feature is e2e-tested:

- [ ] Yes.
- [ ] No, because e2e-tests are not needed.
- [x] No, because e2e-tests will be handled in a follow-up PR

Tests conducted for the PR:

## Codebase

- [x] My code follows the [Effective Go](https://go.dev/doc/effective_go) style guidelines.
- [x] The code was planned and designed following the defined architecture and the separation of concerns.
- [ ] The code has sufficient comments, particularly for all hard-to-understand areas.
- [x] This PR adds value and shows no feature creep.
- [ ] I have augmented the test suite that proves my fix is effective or that my feature works.
- [ ] Adjusted the documentation if the change is user-facing.

### CODE REVIEW

File: internal/otelcollector/config/metric/agent/relabel_configs.go [hunk 1]
```diff
@@ -130,17 +130,17 @@ func dropIfIstioProxy() RelabelConfig {
 	}
 }
 
-func dropIfSchemeAnnotationHTTP(annotated AnnotatedResource) RelabelConfig {
+func dropIfSchemeAnnotationHTTP() RelabelConfig {
```
Comment (a-thaler) [inline internal/otelcollector/config/metric/agent/relabel_configs.go:133]:
```suggestion
func dropIfSchemeHTTP() RelabelConfig {
```

There is no tight coupling to the annotation here

Comment (skhalash) [reply internal/otelcollector/config/metric/agent/relabel_configs.go:133]: Yeah, the name also has to be changed

**Review by a-thaler — APPROVED** [review]

**Pull request merged**